find.lic

```diff
@@ -2,76 +2,109 @@
-    shard_town = [2544, 2550, 2556, 2557, 2558, 2559, 2560, 2561, 2562, 2522, 19_287, 2517, 2516, 2518, 2520, 2521, 2523, 2524, 2525, 2526, 2527, 2528, 2529, 2536, 2537, 2538, 2611, 2503, 2502, 2501, 2496, 2497, 2498, 2511, 2512, 2513, 2514, 2515, 2499, 2500, 2581, 2596, 2575, 2573, 2570, 2567, 2565, 2566, 2564, 2563, 2582, 2585, 2586, 2589, 2597, 2598, 2599, 2600, 2601, 2612, 2607, 2549, 2548, 2547, 2545, 2621, 2544, 2613, 2608, 2617, 2633, 2632, 2631, 2504, 2505, 2506, 2507, 2508, 2509, 2574, 2540, 2539, 2541, 2543, 2603, 2602, 2583, 2584, 2592, 2593, 2594, 2595, 2567, 2571, 2572, 2569, 2568, 2565, 2566, 2564, 2563]
-    pilgrimage_trail = [4179, 4180, 4181, 4182, 4183, 4184, 4185, 4186, 4187, 4188, 4189, 4190, 4191, 4192, 4193, 4194]
+  def initialize
     arg_definitions = [
       [
-        { name: 'npc', regex: /\w+/, description: 'Unique name/noun of the NPC to find' },
-        { name: 'follow', regex: /follow/i, optional: true, description: 'Follow the NPC once it is found' }
+        { name: 'town', options: ['crossing', 'dirge', 'shard', 'boar'], optional: true, description: 'Town/area to search' },
+        { name: 'npc', regex: /^(?!follow)([\w\s\-']+)$/i, optional: true, description: 'Unique name/noun of the NPC to find' },
+        { name: 'follow', regex: /^follow$/i, optional: true, description: 'Follow the NPC once it is found' }
       ]
     ]
 
     args = parse_args(arg_definitions, true)
-    @npc = args.npc
-    @follow = args.follow
 
-    @rooms = case args.npc
-             when /ozursus/i
-               crossing_academy
-             when /shaman/i
-               dirge_town
-             when /theurgist/i
-               shard_town
-             when /monk/i
-               pilgrimage_trail
-             else
-               crossing_town
-             end
-  end
+    # Show help menu if no NPC specified.
+    if args.npc.nil? || args.npc.empty?
+      force_start_script('find', ['help'])
+      exit
+    end
 
-  def initialize
-    setup
+    area = get_town_and_region(args.town, args.npc)
+    rooms = get_rooms(area.town, area.region)
 
-    start_index = @rooms.index(Room.current.id) || 0
-    search(@rooms.rotate(start_index)) unless found?
+    if search?(rooms, args.npc) && args.follow
+      follow(args.npc)
+    end
+  end
 
-    follow if @follow
+  # What geographical area are we searching in?
+  # Use this data to look up the actual rooms to explore.
+  def get_town_and_region(town, npc)
+    town = format_town(town)
+    case npc
+    when /ozursus/i
+      town = 'Crossing' if town.nil? || town.empty?
+      region = 'asemath_academy' if town == 'Crossing'
+    when /shaman/i
+      town = 'Dirge' if town.nil? || town.empty?
+      region = 'rooms'
+    when /theurgist/i
+      town = 'Shard' if town.nil? || town.empty?
+      region = 'rooms'
+    when /monk/i
+      town = 'Boar Clan' if town.nil? || town.empty?
+      region = 'pilgrimage_trail' if town == 'Boar Clan'
+    end
```

Comment:
For backwards compatibility with just specifying an NPC's name without the town/area to search in, this logic does that. It'll set the town automatically unless you specified one in the script args.

Reply:
> Could you update `join-thieves.lic`?

What would you like updated? Are you referring to https://github.com/rpherbig/dr-scripts/blob/master/join-thieves.lic#L12

Technically, `;find beggar` will still work because if no town is specified and if the NPC isn't one of the "special ones" the code already had logic for, then the town will default to Crossing. However, for clarity, we could update the join-thieves script to explicitly pass in "crossing" as the town.

Reply:
`join-thieves` updated per https://github.com/rpherbig/dr-scripts/pull/4957/commits/78ee29318146249e0cbd0861754f9102bff73eb3